src/scope/new/base.js

```diff
@@ -0,0 +1,198 @@
+'use strict'
```

Comment:
Why `/new/base.js` ? Shouldn't it just be "the" scope manager?

Reply:
What do you mean?

Reply:
just curious about the reasoning behind using the word "new" here

Reply:
Because naming is hard :)